src/App.jsx

```diff
@@ -1,86 +1,112 @@
-import {useState, useEffect} from "react";
+import { useState, useEffect } from "react";
 import "./App.css";
 
+const initialTodo = { text: "Iniciar con buena actitud", completed: false };
+
 function App() {
-    const [todos, setTodos] = useState(()=> {
-        const storedTddos = localStorage.getItem('todos');
-        return storedTddos ? JSON.parse(storedTddos) : [{ text: 'Iniciar con buena actitud', completed: false }];
-    });
-    const [newTodo, setNewTodo] = useState('');
+  return <TodoList />;
+}
+
+function TodoList() {
+  const [todos, setTodos] = useState(() => {
+    try {
+      const storedTddos = localStorage.getItem("todos");
+      return storedTddos ? JSON.parse(storedTddos) : [initialTodo];
+    } catch (e) {
+      return [initialTodo];
+    }
```

Comment:
so, we do we need this `try - catch` structure ? 

when reading from local storage, this function: 

```js
JSON.parse(obj)
```

might raise an exception. So we need to catch, and define a "default branch" (`[]` in this case).

two options: keep this logic (as your original code) in the `useState()` initialisation.

```js
const [todos, setTodos] = useState(() => { .... })
```

option no. 2: **initial `useEffect()`**

```js
const [todos, setTodos] = useState([])

useEffect(() => {
  // read logic
}, []) // this runs on component mount
```

any of the two options will work :)